tlslite/utils/cryptomath.py

```diff
@@ -138,8 +138,14 @@ def numberToByteArray(n, howManyBytes=None):
     """
     if howManyBytes == None:
         howManyBytes = numBytes(n)
-    return bytearray((n >> i) & 0xff
-                     for i in range((howManyBytes-1)*8, -1, -8))
+    if endian == "big":
+        return bytearray((n >> i) & 0xff
+                         for i in range((howManyBytes-1)*8, -1, -8))
```

Comment:
nit: for readability, I would use ```reversed(range(...))``` with the one used in the little-endian version.